src/components/elements/image/ImageDisplay.tsx

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2024 Nordeck IT + Consulting GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Skeleton, styled } from '@mui/material';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { ImageElement } from '../../../state';
+import {
+  ElementContextMenu,
+  MoveableElement,
+  SelectableElement,
+  WithSelectionProps,
+} from '../../Whiteboard';
+
+type ImageDisplayProps = ImageElement &
+  WithSelectionProps & {
+    /**
+     * Matrix homeserver base URL
+     */
+    baseUrl: string;
+  };
+
+const Image = styled('image', {
+  shouldForwardProp: (p) => p !== 'loading',
+})<{ loading: boolean }>(({ loading }) => ({
+  userSelect: 'none',
+  WebkitUserSelect: 'none',
+  // prevention of partial rendering if the image has not yet been fully loaded
+  visibility: loading ? 'hidden' : 'visible',
+}));
+
+/**
+ * Render an image.
+ * Show a skeleton of the same while the image is loading.
+ */
+function ImageDisplay({
+  baseUrl,
+  mxc,
+  width,
+  height,
+  position,
+  active,
+  readOnly,
+  elementId,
+}: ImageDisplayProps) {
+  // Image loading errors are dealt with in follow-up tasks.
+  // Don't care about invalid http URLs here.
+  const httpUri = useMemo(
+    () => convertMxcToHttpUrl(mxc, baseUrl) ?? '',
+    [baseUrl, mxc],
+  );
+  const [loading, setLoading] = useState(true);
+  const [showSkeleton, setShowSkeleton] = useState(true);
+
+  useEffect(() => {
+    let timeoutId: ReturnType<typeof setTimeout>;
+
+    if (loading === false && showSkeleton === true) {
+      // keep the skeleton around for some additional time,
+      // so that the image has some time to render
+      timeoutId = setTimeout(() => {
+        setShowSkeleton(false);
+      }, 2500);
+    }
+
+    return () => {
+      clearTimeout(timeoutId);
+    };
+  }, [loading, setShowSkeleton, showSkeleton]);
+
+  const handleLoad = useCallback(() => {
+    setLoading(false);
+  }, [setLoading]);
+
+  const renderedSkeleton = showSkeleton ? (
+    <foreignObject x={position.x} y={position.y} width={width} height={height}>
```

Comment:
I found an issue with this in Safari - the skeleton is rendered with a transformation applied. Size seems to be doubled and a placement transform is applied as well.

<img width="669" alt="Screenshot 2024-03-12 at 17 12 14" src="https://github.com/nordeck/matrix-neoboard/assets/26399/7962530a-8622-4332-b36d-bd35101896a4">

Reply:
Does it happen with every image? Can you provide me the test image?

Reply:
Seems like safari does not scale `foreignObjects` with the SVG canvas: https://bugs.webkit.org/show_bug.cgi?id=23113 :facepalm:

Reply:
Replaced with SVG implementation.